src/resource/solidDataset.test.ts

```diff
@@ -915,6 +917,432 @@ describe("saveSolidDatasetAt", () => {
+      fetch: mockFetch,
+    });
+
+    expect(mockFetch.mock.calls[0][0]).toEqual("https://some.pod/container/");
+  });
+
+  it("sets the right headers to create a Container", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(Promise.resolve(new Response()));
+
+    await createEmptyContainerAt("https://some.pod/container/", {
+      fetch: mockFetch,
+    });
+
+    expect(mockFetch.mock.calls[0][0]).toEqual("https://some.pod/container/");
+    expect(mockFetch.mock.calls[0][1]?.method).toBe("PUT");
+    expect(mockFetch.mock.calls[0][1]?.headers).toHaveProperty(
+      "Link",
+      '<http://www.w3.org/ns/ldp#BasicContainer>; rel="type"'
+    );
+    expect(mockFetch.mock.calls[0][1]?.headers).toHaveProperty(
+      "Content-Type",
+      "text/turtle"
+    );
+    expect(mockFetch.mock.calls[0][1]?.headers).toHaveProperty(
+      "If-None-Match",
+      "*"
+    );
+  });
+
+  it("keeps track of what URL the Container was saved to", async () => {
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(
+        Promise.resolve(
+          mockResponse(undefined, { url: "https://some.pod/container/" })
+        )
+      );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://some.pod/container/",
+      {
+        fetch: mockFetch,
+      }
+    );
+
+    expect(solidDataset.internal_resourceInfo.sourceIri).toBe(
+      "https://some.pod/container/"
+    );
+  });
+
+  it("provides the IRI of the relevant ACL resource, if provided", async () => {
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        mockResponse(undefined, {
+          headers: {
+            Link: '<aclresource.acl>; rel="acl"',
+          },
+          url: "https://some.pod/container/",
+        })
+      )
+    );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://some.pod/container/",
+      { fetch: mockFetch }
+    );
+
+    expect(solidDataset.internal_resourceInfo.aclUrl).toBe(
+      "https://some.pod/container/aclresource.acl"
+    );
+  });
+
+  it("does not provide an IRI to an ACL resource if not provided one by the server", async () => {
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        new Response(undefined, {
+          headers: {
+            Link: '<arbitrary-resource>; rel="not-acl"',
+          },
+        })
+      )
+    );
+
+    const solidDataset = await createEmptyContainerAt(
+      "https://some.pod/container/",
+      { fetch: mockFetch }
+    );
+
+    expect(solidDataset.internal_resourceInfo.aclUrl).toBeUndefined();
+  });
+
+  it("provides the relevant access permissions to the Resource, if available", async () => {
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        new Response(undefined, {
+          headers: {
+            "wac-aLLOW": 'public="read",user="read write append control"',
```

Comment:
Why the casing on `aLLOW` ? Is that to test case-insensitivity ?

Reply:
![image](https://user-images.githubusercontent.com/4251/90506640-e85af900-e154-11ea-812c-7239b5dfc0bf.png)

But yeah, that was the intention, but it was part of the original `getSolidDataset` tests that I copied this from - it's more confusing here, so I'll normalise that.